sky/provision/aws/config.py

```diff
@@ -158,12 +217,12 @@ def _get_role(role_name: str):
             f'Creating new IAM instance profile {colorama.Style.BRIGHT}'
             f'{instance_profile_name}{colorama.Style.RESET_ALL} for use as the '
             'default.')
-        iam.meta.client.create_instance_profile(
-            InstanceProfileName=instance_profile_name)
+        _create_instance_profile_if_not_exists(instance_profile_name)
         profile = _get_instance_profile(instance_profile_name)
         time.sleep(15)  # wait for propagation
     assert profile is not None, 'Failed to create instance profile'
 
+    # TODO(aylei): check if the role associated is the one we expect.
```

Comment:
You already checked in `_ensure_instance_profile_role`. Can we remove the `TODO`?

Reply:
The check in `_ensure_instance_profile_role` only occurs when race happens, i.e. `profile.roles` is empty and the subsequent adding role action encounters `LimitExceeded` error. We still need to check the role in other cases.

I found this TODO is trivial, just added the missing part and removed the TODO.